projects/ad7124-4sdz/src/ad7124-4sdz.c

```diff
@@ -0,0 +1,74 @@
+#include <stdio.h>
+#include <sleep.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <xil_cache.h>
+#include <xparameters.h>
+#include <inttypes.h>
+#include "xil_printf.h"
+#include "ad7124.h"
+#include "ad7124_regs.h"
+#include "spi_extra.h"
+#include "error.h"
+
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+int main(void)
+{
+    struct ad7124_dev *dev; /* A driver handle to pass around */
+    enum ad7124_registers regNr;                       /* Variable to iterate through registers */
+    long timeout = 1000;                               /* Number of tries before a function times out */
+    long ret = 0;                                      /* Return value */
+    long sample;                                       /* Stores raw value read from the ADC */
+ 
+    static struct xil_spi_init_param xil_spi_init_params = {
+		.type = SPI_PS,
+	};
+
+    struct spi_init_param spi_init = {
+    		.platform_ops = &xil_platform_ops,
+			.chip_select = 0,
+			.max_speed_hz = 1000000,
+			.mode = SPI_MODE_3,
+			.extra = &xil_spi_init_params,
+    };
+
+	struct ad7124_init_param ad7124_init_param = {
+		.spi_init = &spi_init,
+		.regs = ad7124_regs,
+		.spi_rdy_poll_cnt = 25000,
+	};
+ 
+    /* Initialize AD7124 device. */
+    ret = ad7124_setup(&dev, &ad7124_init_param);
+    if (ret < 0)
+    	return ret;
+ 
+    /* Read all registers */
+    for (regNr = AD7124_Status; (regNr < AD7124_REG_NO) && !(ret < 0); regNr++)
+    {
+        ret = ad7124_read_register(dev, &ad7124_regs[regNr]);
+        if (ret < 0)
+            return ret;
+    }
+ 
+    while(1) {
+		/* Read data from the ADC */
+		ret = ad7124_wait_for_conv_ready(dev, timeout);
+		if (ret < 0)
+			return ret;
+
+		ret = ad7124_read_data(dev, &sample);
+		if (ret < 0)
+			return ret;
+
+		printf("ADC%"PRIi32" \n", sample);
```

Comment:
is this the only thing that this project should do ? afaik we want IIO support in all projects in the `projects` folder, this PR looks like WIP

Reply:
This pull request was intended for someone who asked a question on Engineering Zone, it is sopose to provide an example on how to build ad7124 driver.